### DISCUSSION TITLE
Verification on Guam grid

### DISCUSSION THREAD

**Discussion by PerryShafran-NOAA** [dtcenter/METplus#1724] [Configuration] [answered]
Hi, there,

Is there any verification option for a grid over Guam, whether it be a mask or NCEP-defined grid or anything you have?  There never was any G104 grid for Guam, so I was curious what the best way to verify a Guam grid would be.

Thanks!

Perry

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @PerryShafran-NOAA, I see you have a question about verifying data over Guam. I took a look at the pre-defined NCEP Grid in [Office Note 388](https://www.nco.ncep.noaa.gov/pmb/docs/on388/tableb.html). I see that grids 175 (10 km lat/lon) and 199 (2.5 km mercator) both mention Guam.

I ran plot_point_obs to visualize these 2 grids in MET:
```
plot_point_obs out/pb2nc/sample_pb.nc G175.ps -plot_grid G175 -title "Grid 175, 10km Lat/Lon"
plot_point_obs out/pb2nc/sample_pb.nc G199.ps -plot_grid G199 -title "Grid 199, 2.5km Mercator"
```
The result is extremely uninteresting... just 2 empty plots. That's because the map data bundled with MET does not include an entry for Guam.
<img width="782" alt="Screen Shot 2022-08-01 at 7 56 56 AM" src="https://user-images.githubusercontent.com/21087144/182164203-309f1c8d-f9c7-460c-886f-34403d19ebc1.png">
 Separately, we've discussed defining map data for plots as shapefiles... which certainly would come in handy here!

Let's switch instead to using these grids in gen_vx_mask. NCEP grid 4 is a global 0.5 degree lat/lon grid. Let's select out the G004 points which are included in these Guam grids by running:
```
gen_vx_mask G004 G175 G004_G175_mask.nc -type grid
gen_vx_mask G004 G199 G004_G199_mask.nc -type grid
```
Here's the result of plotting the output with plot_data_plane:
<img width="904" alt="Screen Shot 2022-08-01 at 8 04 49 AM" src="https://user-images.githubusercontent.com/21087144/182165747-c18939e4-d98f-4900-aaca-21539b927d1a.png">
Looks like the spatial extent of G175 is much larger than G199.

So you could use either of those grids to define a verification mask covering Guam.
Also remember that you are not limited to pre-defined grids. If you have data in a GRIB file covering Guam, you can point to that file and MET will extract and apply the grid definition from that file. In addition, you can manually define a grid string yourself to use whatever grid you'd like.

Hope that helps clarify.

Thanks,
John HG

**Reply by PerryShafran-NOAA** [reply]
Ah, that kind of helps.  Since we are verifying a grib file for Guam, I can use that file to extract the grid definition.  So, to do that, that would mean that I would *not* define a grid or poly line in the config file and let MET use the grid definition of the grid file?  I think that would be preferable if I could do that.  

How would I use that in PB2NC, to define the selection of prepbufr data stations that I would like to use?  And then, how would I use that information in point_stat?  

Thanks for your assistance!

Perry

**Reply by JohnHalleyGotway** [reply]
Perry, correct, MET can just extract the grid definition from that GRIB file centered over Guam.

Let's say that file is named "/path/to/my_guam_data.grib2".

The PB2NC config file includes the `mask` dictionary to filter observations spatially. You'd just set the `grid` to that GRIB file:
```
mask = {
   grid = "/path/to/my_guam_data.grib2";
   poly = "";
}
``` 
That retains only those observations that fall inside that grid.

I tried the same in Point-Stat but that DOES NOT work well:
```
mask = {
   grid  = [ "/path/to/my_guam_data.grib2" ];
   poly  = [];
   sid   = [];
   llpnt = [];
}
```
It errors out... but that doesn't match the details listed in [this section](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html?highlight=mask#mask) of the documentation. So we should either fix that... or just clarify it. We could pretty easily get the grid definition, but the trouble is how we should NAME that masking region in the output.

So instead, you'll need to run gen_vx_mask first:
```
gen_vx_mask /path/to/forecast/file /path/to/my_guam_data.grib2 guam_mask.nc -type grid -name GUAM_MASK
```
And then reference that file in the poly section of the Point-Stat config file:
```
mask = {
   grid  = [];
   poly  = [ "/path/to/guam_mask.nc" ];
   sid   = [];
   llpnt = [];
}
```
Does that work for you?

**Reply by JohnHalleyGotway** [reply]
@PerryShafran-NOAA just wanted to follow up on this discussion. Do you have any more questions about this? If not, please feel free to click the **Mark as answer** button in one of the previous comments.

Thanks!

**Reply by PerryShafran-NOAA** [reply]
Yes, sorry, I was on leave and put this aside for a bit.  

So if I am verifying a Guam forecast file, I can use that very same forecast file to create a mask, but I'll need to use gen_vx_mask to do so.  Is that correct?  

Thanks!

Perry

**Reply by JohnHalleyGotway** [reply]
Perry,

I assume you mean that you want to use your Guam forecast file to define a masking region for some larger grid (e.g. like GFS). If so, then yes, it's pretty easy with gen_vx_mask.

For example, here's a CONUS mask of a GFS grid using sample input files named "gfs.grib2" and "conus.grib1":
```
gen_vx_mask gfs.grib2 conus.grib1 gfs_conus_mask.nc -type grid -name CONUS
DEBUG 1: Input Grid:		gfs.grib2
DEBUG 1: Mask File:		conus.grib1
DEBUG 2: Parsed Input Grid:	LatLon (720 x 361)
DEBUG 2: Parsed Mask Grid:	Lambert Conformal (185 x 129)
DEBUG 1: Output File:		gfs_conus_mask.nc
```
And here's the result via ncview:
<img width="945" alt="Screen Shot 2022-08-15 at 11 02 15 AM" src="https://user-images.githubusercontent.com/21087144/184681090-cffb6f00-5b69-4b34-8c1d-917fce878c23.png">

Note that whatever output variable name you use (i.e. `-name CONUS`) will end up in the VX_MASK column of the resulting .stat output files. So just name it however you'd like that output column to be defined.

**Reply by PerryShafran-NOAA** [reply]
Actually, the only thing I need a Guam mask for is verifying a Guam grid (the Guam RTMA to be exact).  I think that I can use a Guam forecast file once to create the mask, then save the mask in our mask directory and just use that in future Guam verification.

I think I have what I need now, but if I need further help, I'll ask.  

Thanks!

Perry

**Reply by JohnHalleyGotway** [reply]
@PerryShafran-NOAA, ah OK, this is actually much simpler.
If you just want to use all the grid points in the model domain, you can just set:
```
mask = {
   grid = [ "FULL" ];
...
```
The "FULL" grid just means exactly that... use all the input point. But that'll put "FULL" in the VX_MASK output column... and that'll appear downstream in METviewer.

If you'd rather VX_MASK be written as "GUAM", then using a gen_vx_mask file with a variable named "GUAM" would produce the desired result.
